janusgraph-dist/src/test/java/org/janusgraph/pkgtest/AbstractJanusGraphAssemblyIT.java

```diff
@@ -44,115 +26,98 @@
-    protected static final String ZIPFILE_FULL_PATH;
-    protected static final String ZIPFILE_FULL_EXTRACTED;
+import static org.apache.tinkerpop.gremlin.driver.ser.AbstractMessageSerializer.TOKEN_IO_REGISTRIES;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTimeout;
 
-    static {
-        Properties props;
+public abstract class AbstractJanusGraphAssemblyIT extends JanusGraphAssemblyBaseIT {
 
-        try {
-            props = new Properties();
-            java.io.FileReader fr = new FileReader(Joiner.on(File.separator).join(new String[] { "target", "test-classes", "target.properties" }));
-            props.load(fr);
-            fr.close();
-        } catch (IOException e) {
-            throw new AssertionError(e);
-        }
+    public abstract String GetConfigPath();
 
-        BUILD_DIR    = props.getProperty("build.dir");
-        EXPECT_DIR   = props.getProperty("expect.dir");
-        ZIPFILE_PATH = props.getProperty("zipfile.path");
-        ZIPFILE_EXTRACTED = ZIPFILE_PATH.substring(0, ZIPFILE_PATH.length() - 4);
-        ZIPFILE_FULL_PATH = props.getProperty("zipfile-full.path");
-        ZIPFILE_FULL_EXTRACTED = ZIPFILE_FULL_PATH.substring(0, ZIPFILE_FULL_PATH.length() - 4);
+    public abstract String GetServerConfigPath();
 
-        Properties p = new Properties();
-        p.put("file.resource.loader.path", EXPECT_DIR);
-        Velocity.init(p);
-    }
+    public abstract String GetGraphName();
 
-    protected void testSimpleGremlinSession(String graphConfig, String graphToString, boolean full) throws Exception {
-        unzipAndRunExpect("single-vertex.expect.vm", graphConfig, graphToString, full, false);
+    public Boolean SupportsIndex() {
+        return false;
     }
 
-    private static boolean serverListening(String host, int port)
-    {
-        Socket s = null;
-        try
-        {
-            s = new Socket(host, port);
-            return true;
-        }
-        catch (Exception e)
-        {
-            return false;
-        }
-        finally
-        {
-            if(s != null)
-                try {s.close();}
-                catch(Exception e){}
-        }
+    @Test
+    public void testSimpleGremlinSession() throws Exception {
+        unzipAndRunExpect("single-vertex.expect.vm", GetConfigPath(), GetGraphName(), false, false);
     }
 
-    protected void testJanusGraphServer(String janusgraphServerConfig, boolean full) throws Exception {
-        final boolean debug = false;
-        ImmutableMap<String, String> contextVars = ImmutableMap.of("janusgraphServerConfig", janusgraphServerConfig);
-        unzipAndRunExpect("janusgraph-server-sh.before.expect.vm", contextVars, full, debug);
-        assertTimeout(Duration.ofSeconds(30), () -> {
-            while (!serverListening("localhost",8182)) {
-                Thread.sleep(1000);
-            }
-        });
+    @Test
+    @Tag(TestCategory.FULL_TESTS)
```

Comment:
Question: what is the purpose of this tag and how does it work?

Reply:
This tag is just a preparation for the java 11 to filter out full tests.